### PR TITLE
Adding timing messages for verbosity level.

### PR DESCRIPTION
**Pull request by radumereuta**
@dwightguth please review. Here is a sample output:

```
Parse command line options                                   =    18
Importing: D:\work\test\test.k
Importing: D:\work\k\k-distribution\include\builtin\kast.k
Importing: D:\work\test\test.k
Importing: D:\work\k\k-distribution\include\builtin\domains.k
Parse definition [0/62 rules]                                =  1560
Apply compile pipeline                                       =   126
Save to disk                                                 =    64
Cleanup                                                      =     0
Total                                                        =  1768
```

### CODE REVIEW

**Comment by dwightguth** [comment]
Instead of modifying every call site to add an object that is only used by --verbose, even though --verbose isn't being specified, can you add a constructor to Kompile that doesn't require that argument?

**Comment by radumereuta** [comment]
Is this what you had in mind?

File: kernel/src/main/java/org/kframework/utils/errorsystem/KExceptionManager.java [hunk 1]
```diff
@@ -24,6 +24,10 @@
 public class KExceptionManager {
     private final List<KException> exceptions = Collections.synchronizedList(new ArrayList<>());
 
+    public GlobalOptions getOptions() {
```
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/utils/errorsystem/KExceptionManager.java]: don'

Comment (dwightguth) [inline kernel/src/main/java/org/kframework/utils/errorsystem/KExceptionManager.java]: This is a bad way of accessing GlobalOptions. Get it from KompileOptions.global

File: k-distribution/src/test/java/org/kframework/Konsole.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import org.kframework.kore.K;
 import org.kframework.main.GlobalOptions;
 import org.kframework.tiny.Rewriter;
+import org.kframework.utils.Stopwatch;
```
Comment (dwightguth) [inline k-distribution/src/test/java/org/kframework/Konsole.java]: revert these three files

**Comment by radumereuta** [comment]
Did I get it right this time?

**Comment by radumereuta** [comment]
Jenkins retest this please

File: kernel/src/main/java/org/kframework/kompile/Kompile.java [hunk 3]
```diff
@@ -81,19 +82,34 @@
...
+        this.kem = kem;
+        this.kompileOptions = kompileOptions;
+        this.parser = new ParserUtils(files, kem, kompileOptions.global);
+        this.cacheParses = cacheParses;
+        this.loader = new BinaryLoader(kem);
+        this.sw = sw;
+    }
 
     public Kompile(KompileOptions kompileOptions, FileUtil files, KExceptionManager kem, boolean cacheParses) {
         this.files = files;
```
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/kompile/Kompile.java]: can you make this constructor this() the one above?

File: kernel/src/main/java/org/kframework/utils/errorsystem/KExceptionManager.java [hunk 4]
```diff
@@ -101,7 +101,17 @@ public void addKException(KException kex) {
     }
 
     public void addAllKException(Collection<KException> kex) {
-        exceptions.addAll(kex);
+        if (this.options.warnings == GlobalOptions.Warnings.NONE)
```
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/utils/errorsystem/KExceptionManager.java]: take a look at how registerInternal works. You shouldn't need to have any kind of if/else clause, just do a filter on the method that that method calls.

**Comment by radumereuta** [comment]
Tests passed, and I think I've addressed all the comments.

**Pull request merged**